pennylane/fourier/simple_spectrum.py

```diff
@@ -267,19 +199,28 @@ def wrapper(*args, **kwargs):
 
                 if len(op.parameters) != 1:
                     raise ValueError(
-                        "Can only consider one-parameter gates as data-encoding gates; "
-                        f"got {op.name}."
+                        "Can only consider one-parameter gates as "
+                        f"data-encoding gates; got {op.name}."
                     )
 
-                spec = _get_spectrum(op)
+                spec = _get_spectrum(op, decimals=decimals)
 
-                # if id has been seen before,
-                # join this spectrum to another one
+                # if id has been seen before, join this spectrum to another one
                 if id in freqs:
                     spec = _join_spectra(freqs[id], spec)
 
                 freqs[id] = spec
 
+        # Turn spectra into sorted lists and include negative frequencies
+        for id, spec in freqs.items():
+            spec = sorted(spec)
+            freqs[id] = [-f for f in spec[:0:-1]] + spec
+
+        # Add trivial spectrum for requested gate ids that are not in the circuit
+        if encoding_gates is not None:
+            for id in set(encoding_gates).difference(freqs):
+                freqs[id] = []
```

Comment:
Here the requested `id`s that are not in the circuit are added. The spectrum `[]` only ever is given to such `id`s.